integration/spark2/src/main/java/org/apache/carbondata/spark/vectorreader/VectorizedCarbonRecordReader.java

```diff
@@ -144,14 +145,10 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
       }
       throw new InterruptedException(e.getMessage());
     } catch (Exception e) {
-      Throwable ext = e;
-      while (ext != null) {
-        if (ext instanceof FileNotFoundException) {
-          LOGGER.error(e);
-          throw new InterruptedException(
-              "Insert overwrite may be in progress.Please check " + e.getMessage());
-        }
-        ext = ext.getCause();
+      if (ExceptionUtils.hasCause(e, FileNotFoundException.class)) {
```

Comment:
hasCause is not recursive, so can use indexOfThrowable

Reply:
fixed